mm2src/mm2_main/src/lp_swap/maker_swap.rs

```diff
@@ -576,6 +576,26 @@ impl MakerSwap {
             },
         };
 
+        // Validate maker_coin_htlc_pubkey realness
+        match self.taker_coin.validate_other_pubkey(taker_data.maker_coin_htlc_pub()) {
+            Ok(_) => (),
+            Err(err) => {
+                return Ok((Some(MakerSwapCommand::Finish), vec![MakerSwapEvent::NegotiateFailed(
+                    ERRL!("!taker_data.maker_coin_htlc_pub {}", err).into(),
+                )]))
+            },
+        };
+
+        // Validate taker_coin_htlc_pubkey realness
+        match self.taker_coin.validate_other_pubkey(taker_data.taker_coin_htlc_pub()) {
```

Comment:
Here's correct, our `taker_coin` should validate the other's `taker_coin_htlc_pub`.